File: old/de/hglabor/commands/BingoCommand.java
package de.hglabor.commands;

import de.hglabor.Bingo;
import de.hglabor.utils.BingoManager;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import java.util.Locale;

public class BingoCommand implements CommandExecutor {
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if(!(sender instanceof Player)) {
            sender.sendMessage("§9KEIN SPIELER");
        } else {
            Player player = (Player) sender;
            Inventory inventory;
            if(player.getLocale().toLowerCase().contains("de")) {
                inventory = Bukkit.createInventory(null, 18, Bingo.getLocalization().getFieldName(Locale.GERMAN));
            } else {
                inventory = Bukkit.createInventory(null, 18, Bingo.getLocalization().getFieldName(Locale.ENGLISH));
            }
            int slot = -1;
            for(Material material : BingoManager.getNeededMaterials()) {
                slot+=1;
                ItemStack stack = new ItemStack(material);
                ItemMeta meta = stack.getItemMeta();
                meta.addItemFlags(ItemFlag.HIDE_ENCHANTS);
                if(BingoManager.hasChecked(player, material)) {
                    meta.addEnchant(Enchantment.KNOCKBACK, 10, true);
                }
                stack.setItemMeta(meta);
                inventory.setItem(slot, stack);
            }
            player.openInventory(inventory);
        }
        return false;
    }
}

File: src/main/kotlin/de/hglabor/commands/BingoCommand.kt
package de.hglabor.commands

import de.hglabor.core.GamePhaseManager
import de.hglabor.core.mechanics.MaterialManager
import de.hglabor.core.phase.InGamePhase
import de.hglabor.utils.hasChecked
import de.hglabor.utils.sendMsg
import net.axay.kspigot.chat.KColors
import net.axay.kspigot.commands.command
import net.axay.kspigot.commands.runs
import net.axay.kspigot.items.itemStack
import net.axay.kspigot.items.meta
import net.axay.kspigot.items.name
import net.axay.kspigot.utils.mark
import net.kyori.adventure.text.Component
import org.bukkit.Bukkit
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemFlag

object BingoCommand {
    init {
        command("bingo") {
            runs {
                if (GamePhaseManager.phase !is InGamePhase) {
                    player.sendMsg("Diese Command kannst du jetzt nicht ausführen")
                }
                val inventory = Bukkit.createInventory(
                    null,
                    5 * 9,
                    Component.text("${KColors.CORNFLOWERBLUE}Bingo")
                )
                MaterialManager.materials.forEach {
                    val itemStack = itemStack(it) {
                        meta {
                            addItemFlags(ItemFlag.HIDE_ATTRIBUTES)
                            addItemFlags(ItemFlag.HIDE_POTION_EFFECTS)
                            addItemFlags(ItemFlag.HIDE_ENCHANTS)
                            name = "${KColors.CORNFLOWERBLUE}${it.name.lowercase().replace("_", " ")}"
                            //TODO onClick -> show item recipe / if item is craftable (configurable)
                            if (player.hasChecked(it)) {
                                addEnchant(Enchantment.PROTECTION_FALL, 1, true)
                            }
                        }
                    }
                    itemStack.mark("locked")
                    inventory.addItem(itemStack)
                }
                player.openInventory(inventory)
            }
        }
    }
}

